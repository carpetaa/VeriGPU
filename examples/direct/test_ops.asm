li x2, 7
li x3, 11

slt x1, x2, x3
outr x1  ; should be 1

slt x1, x3, x2
outr x1  ; should be 0

slt x1, x2, x2
outr x1  ; should be 0

sltu x1, x0, x2
outr x1  ; should be 1

li x2, 0
sltu x1, x0, x2
outr x1  ; should be 0

li x2, -1
sltu x1, x0, x2
outr x1  ; should be 1

li x2, 0b1011
li x3, 0b0110
and x1, x2, x3
outr x1  ; 0b0010

# try add, sll, srl

li x1 15
li x2 8
add x3, x1, x2
outr x3

li x2 1
sll x3, x1, x2
outr x3

li x2 2
sll x3, x1, x2
outr x3

li x2 3
sll x3, x1, x2
outr x3

li x2 1
srl x3, x1, x2
outr x3

li x2 2
srl x3, x1, x2
outr x3

li x2 3
srl x3, x1, x2
outr x3

or x3, x1, x2
outr x3

xor x3, x1, x2
outr x3

sub x3, x2, x1
outr x3

li x1 -150
li x2 3
sra x3, x1, x2
outr x3

halt
