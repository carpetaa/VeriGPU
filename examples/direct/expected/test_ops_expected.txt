out 0 00000000000000000000000000000001 00000001 1
out 1 00000000000000000000000000000000 00000000 0
out 2 00000000000000000000000000000000 00000000 0
out 3 00000000000000000000000000000001 00000001 1
out 4 00000000000000000000000000000000 00000000 0
out 5 00000000000000000000000000000001 00000001 1
out 6 00000000000000000000000000000010 00000002 2
out 7 00000000000000000000000000010111 00000017 23
out 8 00000000000000000000000000011110 0000001e 30
out 9 00000000000000000000000000111100 0000003c 60
out 10 00000000000000000000000001111000 00000078 120
out 11 00000000000000000000000000000111 00000007 7
out 12 00000000000000000000000000000011 00000003 3
out 13 00000000000000000000000000000001 00000001 1
out 14 00000000000000000000000000001111 0000000f 15
out 15 00000000000000000000000000001100 0000000c 12
out 16 11111111111111111111111111110100 fffffff4 4294967284
out 17 11111111111111111111111111101101 ffffffed 4294967277